src/App.jsx

```diff
@@ -14,33 +16,106 @@ export const goodsFromServer = [
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const INITIAL_STATE = { sort: 'none', reverse: false };
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const ALPHABET_SORT = 'alphabet';
+const STR_LENGTH_SORT = 'length';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+const getPreparedGoods = (goods, { sort, reverse }) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sort) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sort) {
+        case ALPHABET_SORT:
+          return good1.localeCompare(good2);
+
+        case STR_LENGTH_SORT:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState(INITIAL_STATE);
+
+  const handleSort = sortType => {
+    setSortField(prevState => ({
+      sort: sortType,
+      reverse: prevState.reverse,
+    }));
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleReset = () => setSortField(INITIAL_STATE);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField.sort !== ALPHABET_SORT,
+          })}
+          onClick={() => handleSort(ALPHABET_SORT)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField.sort !== STR_LENGTH_SORT,
+          })}
+          onClick={() => handleSort(STR_LENGTH_SORT)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !sortField.reverse,
+          })}
+          onClick={() =>
+            setSortField({ ...sortField, reverse: !sortField.reverse })
+          }
+        >
+          Reverse
+        </button>
+
+        {sortField.sort === 'none' && sortField.reverse === false ? (
+          ''
+        ) : (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => handleReset()}
+            hidden
```

Comment:
The `hidden` attribute on the 'Reset' button makes it always invisible. Consider removing this attribute to allow the button to be visible when sorting or reversing is active, as intended by the surrounding conditional logic.